ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/EvaluatorImpl.java

```diff
@@ -0,0 +1,121 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.shell;
+
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.shell.exceptions.BallerinaShellException;
+import io.ballerina.shell.exceptions.PreprocessorException;
+import io.ballerina.shell.invoker.ShellSnippetsInvoker;
+import io.ballerina.shell.parser.TreeParser;
+import io.ballerina.shell.preprocessor.Preprocessor;
+import io.ballerina.shell.snippet.Snippet;
+import io.ballerina.shell.snippet.factory.SnippetFactory;
+import io.ballerina.shell.utils.StringUtils;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Optional;
+
+/**
+ * Implemented Evaluator.
+ *
+ * @since 2.0.0
+ */
+class EvaluatorImpl extends Evaluator {
+    protected EvaluatorImpl(Preprocessor preprocessor, TreeParser treeParser,
+                            SnippetFactory snippetFactory, ShellSnippetsInvoker invoker) {
+        super(preprocessor, treeParser, snippetFactory, invoker);
+    }
+
+    @Override
+    public void initialize() throws BallerinaShellException {
+        try {
+            invoker.initialize();
+        } catch (BallerinaShellException e) {
+            addAllDiagnostics(invoker.diagnostics());
+            throw e;
+        }
+    }
+
+    @Override
+    public String evaluate(String source) throws BallerinaShellException {
+        try {
+            Collection<String> statements = preprocessor.process(source);
+            Collection<Node> nodes = treeParser.parse(statements);
+            Collection<Snippet> snippets = snippetFactory.createSnippets(nodes);
+            Optional<Object> invokerOut = invoker.execute(snippets);
+            return invokerOut.map(StringUtils::getExpressionStringValue).orElse(null);
+        } finally {
+            addAllDiagnostics(preprocessor.diagnostics());
+            addAllDiagnostics(treeParser.diagnostics());
+            addAllDiagnostics(snippetFactory.diagnostics());
+            addAllDiagnostics(invoker.diagnostics());
+            preprocessor.resetDiagnostics();
+            treeParser.resetDiagnostics();
+            snippetFactory.resetDiagnostics();
+            invoker.resetDiagnostics();
+        }
+    }
+
+    @Override
+    public void evaluateDeclarationFile(String filePath) throws BallerinaShellException {
+        try {
+            String statements = Files.readString(Paths.get(filePath), Charset.defaultCharset());
+            Collection<Node> nodes = treeParser.parseDeclarations(statements);
+            Collection<Snippet> snippets = snippetFactory.createSnippets(nodes);
+            invoker.execute(snippets);
+        } catch (IOException e) {
+            addErrorDiagnostic("File open failed: " + filePath);
```

Comment:
How about
```suggestion
            addErrorDiagnostic("Error loading from file '" + filePath + "'");
```

Can we add some information from the exception as to what exactly went wrong?